### PR TITLE
Log failed and successful payments

### LINKED ISSUE #193: Store information about failed payments in the database
Store information about failed payments in the database so that the issue can be investigated by admin

Note: only work with StripeCLI running

Split to #124

### LINKED ISSUE #197: Create Payments table
Create a payments table to hold all successful payments

### PR DESCRIPTION
**Pull request by SamWinterhalder**
- Created `Payment` model
- Successful payments are logged in this table
- Added `last_payment_date` field to `Membership` model - this is updated on successful payment
<br>

- Created `FailedPayment` model
- Failed payments are logged in the table for admin to look over
<br>
Fixes #197
Fixes #193

### CODE REVIEW

**Comment by SamWinterhalder** [comment]
This seems to be adding the changes in #194 
Probably an error when I created this branch

File: memberships/views.py [hunk 1]
```diff
@@ -179,6 +181,24 @@ def stripe_webhook(request):
             json.loads(request.body), settings.STRIPE_SECRET_KEY
         )
         stripe_webhook = StripeWebhook()
+        if event["type"] == "invoice.payment_failed":
+            f_payment = FailedPayment.objects.create(
+                stripe_customer_id=event["data"]["object"]["customer"],
+                stripe_subscription_id=event["data"]["object"]["subscription"],
+                stripe_event_type=event["type"],
+            )
+        if event["type"] == "invoice.payment_succeeded":
```
Comment (CarwynNelson) [inline memberships/views.py:190]: Could you write a test case for this `payment_succeeded` scenario?

**Review by CarwynNelson — CHANGES_REQUESTED** [review]
Great work :)

Only a comment about adding a test case, but everything else looks ready to merge.

File: memberships/views.py (hunk 1 above)
Comment (SamWinterhalder) [reply memberships/views.py:190]: Of course

**Review by CarwynNelson — APPROVED** [review]
Looks great, thanks for this :)

In the near-term I think we should either:
1. Move the `if event==` stuff to the `StripeWebhook` class as that is what I had intended it for
2. Just delete the `StripeWebhook` class.

I'm not a massive fan of the `StripeWebhook` class but IMHO it's the cleanest way we have so far, so any cleaner ways you can think to do it are welcome :)

**Pull request merged**